src/Compilers/CSharp/Portable/Symbols/TypedConstantExtensions.cs

```diff
@@ -140,7 +138,10 @@ private static string DisplayUnsignedEnumConstant(TypedConstant constant, Specia
             }
 
             // Unable to decode the enum constant, just display the integral value
-            return constant.ValueInternal.ToString();
+            Debug.Assert(constant.ValueInternal is object);
```

Comment:
>Debug.Assert(constant.ValueInternal is object) [](start = 12, length = 46)

Why assert rather than using `!`?

Reply:
Because this is the result of a check in a calling method (`if (constant.IsNull)` in `ToCSharpString`), but not in the present method.

---
In reply to: [443063994](https://github.com/dotnet/roslyn/pull/44266#discussion_r443063994) [](ancestors = 443063994)